### PR TITLE
feat: signal option

### PR DESCRIPTION
**Pull request by PierreDemailly**

### CODE REVIEW

File: src/prompts/abstract.ts [hunk 1]
```diff
@@ -6,6 +6,8 @@ import { Writable } from "node:stream";
 // Import Internal Dependencies
 import { stripAnsi } from "../utils.js";
 import { PromptAgent } from "../prompt-agent.js";
+import { TimeoutError } from "../errors/timeout.js";
+import EventEmitter from "node:events";
```
Comment (fraxken) [inline src/prompts/abstract.ts:10]: Node.js

File: src/prompts/abstract.ts [hunk 2]
```diff
@@ -15,20 +17,37 @@ type Stdout = NodeJS.WriteStream & {
   fd: 1;
 }
 
-export class AbstractPrompt<T> {
+export interface AbstractPromptOptions {
+  stdin?: Stdin;
+  stdout?: Stdout;
+  message: string;
+  timeout?: number;
```
Comment (fraxken) [inline src/prompts/abstract.ts:24]: Why not using an AbortSignal instead of number?

Comment (PierreDemailly) [reply src/prompts/abstract.ts:24]: I planned to have both options `signal` & `timeout`.
Do you think we should only keep signal ?

Comment (fraxken) [reply src/prompts/abstract.ts:24]: Since we can do `AbortSignal.timeout(1000)` I think yes.

**Review by fraxken — APPROVED** [review]

**Pull request merged**